### ISSUE TITLE
Where are `.JuliaFormatter.toml` files searched for?

### ISSUE THREAD

**Issue by simsurace** [julia-vscode/julia-vscode#3560]
The docs are slightly unclear about this. They say in the workspace but what if there are several folders? I do have a repo with a toml file in one of my workspaces but the style is not respected --- formatting instead is done with the default very light setting.

**Comment by IanButterworth** [comment]
See https://github.com/domluna/JuliaFormatter.jl/issues/771 for a pain point

**Comment by pfitzseb** [comment]
>  They say in the workspace but what if there are several folders?

In *any* workspace. Basically, if you can open the file from the file explorer in VS Code, we should be able to pick find it.

**Comment by IanButterworth** [comment]
I'm not sure I understand the rationale for the current design. AFAIU`.JuliaFormatter.toml` files are for specifying directory-specific formatting rules, so why would vscode not respect that? 

Is it just that it's expected that people only have one folder (repo) open per workspace? I generally have many open.

**Comment by simsurace** [comment]
IIUC workspace is the top level in the VSCode explorer. If I have multiple folders open, does the extension look for the TOML in the same folder first?

**Comment by pfitzseb** [comment]
It doesn't matter how many workspace folders you have open. We're following the same behaviour as JuliaFormatter, except for never looking at the global config and not veering outside the bounds of the directory you have opened in VS Code (so the theoretically obvious way for adding a global config by creating a `/.JuliaFormatter.toml` also doesn't work).

The main motivation for that is that all format options need to be inside of your repo for reproducibility (otherwise formatting ends up flipflopping between N different configs). That's a decision we (I?) made a couple years back, but certainly nothing we couldn't revisit.

**Comment by IanButterworth** [comment]
> not veering outside the bounds of the directory you have opened in VS Code

What if you have multiple directories open?

To be concrete, say I have these folders open in a single workspace:

- `julia`
- `MyFoo`

where `julia` doesn't use JuliaFormatter, so formatting on save isn't wanted, but `MyFoo` does, so formatting on save is wanted.

Currently the system doesn't accommodate this, as far as I know?

**Comment by IanButterworth** [comment]
To clarify, what I mean is that say `julia` had a `.JuliaFormatter` that disabled JuliaFormatter (which is what I've considered adding to julia)

**Comment by IanButterworth** [comment]
I confirm the bug then.

Saving `.JuliaFormatter.toml` in the root of the julia repo with
```
ignore = [""]
```
 then saving say `base/loading.jl` will format that file.

**Comment by iago-lito** [comment]
I'm not sure I exactly confirm this, but since this issue was opened yesterday, I think it's the right place/time to report that julia formatter seems to have stopped respecting the `.JuliaFormatter.toml` sitting next to my `Project.toml` at the root of my package folder.. this week.
Has anything changed recently regarding this?

**Comment by pfitzseb** [comment]
Yes, it is. Are you using the LS in single-file mode? If so, that'd explain the issue (since we're only searching in the workspace dir).
I don't really remember my original reasoning for that, but I'm down to change this.

**Comment by svretina** [comment]
Emacs initializes a project (and sees the parent directory) and activates the LS. I don't know what that would mean for your question. As a current fix I just copied my `.JuliaFormatter.toml` into my `/src`. 
Happy to hear that this will change. 
Should this be moved as a bug issue in LS.jl instead?
